src/Compilers/CSharp/Portable/BoundTree/BoundNodes.xml

```diff
@@ -427,12 +431,19 @@
     <Field Name="GetResult" Type="MethodSymbol" Null="allow"/>
   </Node>
 
-  <Node Name="BoundTypeOfOperator" Base="BoundExpression">
+  <AbstractNode Name="BoundTypeOf" Base="BoundExpression">
     <!-- Non-null type is required for this node kind -->
     <Field Name="Type" Type="TypeSymbol" Override="true" Null="disallow"/>
-    <Field Name="SourceType" Type="BoundTypeExpression"/>
     <!-- Well-known member populated during lowering -->
     <Field Name="GetTypeFromHandle" Type="MethodSymbol" Null="allow"/>
+  </AbstractNode>
+    
+  <Node Name="BoundTypeOfOperator" Base="BoundTypeOf">
+    <Field Name="SourceType" Type="BoundTypeExpression"/>
+  </Node>
+    
+  <Node Name="BoundTypeReferenceTypeOfOperator" Base="BoundTypeOf">
```

Comment:
It looks like the only reason we have this node is to be able to express desire to load System.Type instance for PrivateImplementationDetails type. I suggest to create a node just for that, without generalizing and adding ability to target any Cci.ITypeReference. Here is the proposal:
- remove BoundTypeReferenceExpression node
- remove SourceType field from BoundTypeReferenceTypeOfOperator
- rename BoundTypeReferenceTypeOfOperator to BoundTypeTypeOfOfPrivateImplementationDetails

CodeGenerator has instance of PEModuleBuilder and it will be able to get the Cci.ITypeReference from it.